subprojects/publish/src/main/java/org/gradle/api/publish/internal/metadata/ModuleMetadataJsonWriter.java

```diff
@@ -265,9 +265,9 @@ private void writeDependencyConstraints(List<ModuleMetadataSpec.DependencyConstr
         writeArray("dependencyConstraints", () -> {
             for (ModuleMetadataSpec.DependencyConstraint constraint : constraints) {
                 writeObject(() -> {
-                    write("group", constraint.group);
-                    write("module", constraint.module);
-                    writeVersionConstraint(constraint.version);
+                    write("group", constraint.coordinates.group);
+                    write("module", constraint.coordinates.name);
+                    writeVersionConstraint(constraint.coordinates.version);
```

Comment:
❓ Does this change behavior? What's the difference between `constraint.group` and `constraint.coordinates.group`?

Reply:
Nevermind, the next file removes `group` and adds `coordinates`.